apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Snippets/ErrorReportingSnippets.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Cloud.ClientTesting;
+using Google.Cloud.Diagnostics.Common.IntegrationTests;
+using Google.Cloud.Diagnostics.Common.Tests;
+using Google.Cloud.ErrorReporting.V1Beta1;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Google.Cloud.Diagnostics.AspNetCore.Snippets
+{
+    public class ErrorReportingSnippetsTests : IDisposable
+    {
+        private static readonly bool _isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+
+        private readonly string _testId;
+        private readonly ErrorEventEntryPolling _polling;
+
+        private readonly TestServer _server;
+        private readonly HttpClient _client;
+
+        private readonly DateTime _startTime;
+
+        public ErrorReportingSnippetsTests()
+        {
+            _polling = new ErrorEventEntryPolling();
```

Comment:
Is this something that exists already? (I'm not very familiar with the diagnostics bits.)

Reply:
Yes, it's defined and used on Google.Cloud.Diagnostics.Common.IntegrationTests. I'm reusing as much as I can from IntegrationTests and even copy pasting and changing the code I cannot reuse as is.
But actually I think that with minimum tweaking this class in particular can be made part of the public client lib? It seems like a nice helper class to offer our users. What do you think @jskeet @iantalarico , new issue for a separate PR?

Reply:
Definitely a separate PR, and I'd want to think about it carefully. I'm a big fan of a small public API surface - less to version!

Reply:
Agreed with thinking it over. Created issue #2200 to keep track.